Enhance search in Checkboxes (Tree) interface

gitstart opened the pull request:
<!--

Heya! Thanks for opening a Pull Request! If your PR is implementing a new feature or fix for Directus, please make sure your PR adheres to the following requirements:

- The PR closes an Issue (not Discussion)
- Tests are added/updated and are passing locally if applicable
- Documentation was added/updated if applicable

Please make sure to "Link" the issue you're closing. Without a Linked issue, this PR won't be accepted. See https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue for more information.

-->

Fixes #18918

Demo: https://www.loom.com/share/8f5a711e0ade4c1ba03c6f30d8ec15e2

gitstart committed:
Make v-checkbox-tree search accent sensitive

Co-authored-by: Thiago Nascimbeni <tnascimbeni@gmail.com>

gitstart committed:
Make v-checkbox-tree search accent sensitive

Co-authored-by: v1b3m <vibenjamin6@gmail.com>
Co-authored-by: Thiago Nascimbeni <tnascimbeni@gmail.com>

paescuj committed:
Update use-visible-children

paescuj committed:
Switch to modern-diacritics, diacritics is outdated

paescuj commented on .changeset/dirty-shrimps-laugh.md:
```diff
@@ -0,0 +1,5 @@
+'@directus/app': patch
+---
+
+Make v-checkbox-tree search accent sensitive
```
```suggestion
Enabled accent sensitive search in checkbox tree component 
```

paescuj committed:
Update .changeset/dirty-shrimps-laugh.md

paescuj committed:
Forgot to update v-highlight 😇

DanielBiegler reviewed (changes requested):
I think the highlighting change broke something because on 0baab84c36d0a91fe4dca13437369eddc8a02dd4 it works:

![image](https://github.com/directus/directus/assets/14810858/889f76a3-4de7-484e-879c-5707930f8e74)

But then on 349db6ec7d894af548e3ad6237a697b3d8306641 it breaks:

![image](https://github.com/directus/directus/assets/14810858/f0174a7f-9942-4806-ac1a-2e66c6102b15)

Also is this behaviour a separate issue or related?:

[Screencast from 18.08.2023 19:18:34.webm](https://github.com/directus/directus/assets/14810858/1848d27f-bee2-4b18-8165-55b192979149)

paescuj commented:
> I think the highlighting change broke something because on [0baab84](https://github.com/directus/directus/commit/0baab84c36d0a91fe4dca13437369eddc8a02dd4) it works:
> 
> ![image](https://user-images.githubusercontent.com/14810858/261666335-889f76a3-4de7-484e-879c-5707930f8e74.png)
> 
> But then on [349db6e](https://github.com/directus/directus/commit/349db6ec7d894af548e3ad6237a697b3d8306641) it breaks:
> 
> ![image](https://user-images.githubusercontent.com/14810858/261666948-f0174a7f-9942-4806-ac1a-2e66c6102b15.png)

Looking into it 👍 

> Also is this behaviour a separate issue or related?:
> Screencast.from.18.08.2023.19.18.34.webm

I think that has already been the case before.

rijkvanzanten commented on app/package.json:
```diff
@@ -138,6 +136,7 @@
 		"micromustache": "8.0.3",
 		"mime": "3.0.0",
 		"mitt": "3.0.0",
+		"modern-diacritics": "2.3.1",
```
I have my doubts about this packages. No downloads, no indication of maintenance. I know the original diacritics hasn't been updated in what feels like centuries, but it still has a huge amount of people checking/relying on it 🤔

paescuj replied:
>  I know the original diacritics hasn't been updated in what feels like centuries, but it still has a huge amount of people checking/relying on it 🤔

That's sums up the problem of the Node.js ecosystem quite well, if you ask me 🤣 People just don't care about outdated packages, so I'm not sure whether this number should be that decisive. I took a glance at `modern-diacritics` and it made a solid impression to me. If for example a security bug should appear, I also think that the chances are higher that the newer package will receive a fix than the package that has not received any updates for 7 years.
But I get your doubts, I'm always checking number of downloads too!

![Screenshot 2023-08-18 at 20-16-17 diacritics vs fold-to-ascii vs modern-diacritics vs normalize-diacritics npm trends](https://github.com/directus/directus/assets/5363448/dc9298e8-5e58-4b6a-9abf-386eae285663)
https://npmtrends.com/diacritics-vs-fold-to-ascii-vs-modern-diacritics-vs-normalize-diacritics

We could also take a look at [`normalize-diacritics`](https://www.npmjs.com/package/normalize-diacritics
).

Let me know what you prefer!

rijkvanzanten replied:
A third option is: this library really doesn't do anything from a find and replace on a list of hardcoded diacritics characters. Almost wondering if it makes more sense to just roll that ourselves and remove the dependency on third party software altogether

paescuj replied:
Right, also played with this thought! 👍

paescuj replied:
Will be addressed in a separate PR 👍

paescuj committed:
Fix everything :-)

- padding in v-highlight to prevent overflow
- normalize not only search value but also item value to get all results
- open tree when search matches child element
- don't let search override open selection (restores selection when
  clearing search)

DanielBiegler commented:
@paescuj my brother in christ, go to sleep lmaooo

paescuj committed:
Update snapshot

paescuj committed:
Update changeset

paescuj commented:
Love the name of the changeset btw. 🦐😆

DanielBiegler reviewed (approved):
Looks really nice! 

[Screencast from 21.08.2023 12:11:26.webm](https://github.com/directus/directus/assets/14810858/6b4b3f9d-dac7-4e42-a759-39cd36f68d42)

paescuj committed:
Merge branch 'main' into DIRECTUS-OSS-30

paescuj committed:
Switch back to `diacritics` for now, will be replaced in separate PR

paescuj committed:
Export `Choice` type to fix TS issue

> Default export of the module has or is using private name 'Choice'.ts(4082)

paescuj committed:
Require text and value, otherwise select & search doesn't work
